### DISCUSSION TITLE
[SONOFF Dual R2] Long press feature

### DISCUSSION THREAD

**Discussion by dokho33** [arendst/Tasmota#10782] [Q&A] [answered]
Hi !

Sorry for my english, I'm French ;)

Is it possible to use long press function on sonoff dual r2 (toggle relay only with a long press and nothing on short press) ?

I've try setoption114 to detach switch from relay but it doesn't work

Regards,

**Comment by barbudor** [comment, marked as answer]
Bonjour

Yes it is but you need a switchmode that supports Hold such as `switchmode 5`
And implement a rule `on switch1#state=3 do power1 toggle`

**Reply by dokho33** [reply]
Hello ;)

I've try this but setoption114 don't detach switch from relay. So short press toggle relay too.

I would like relays was toggled only by long press.

I know how to set switchmode, make rule to publish on topic,..., my only problem is detach switch from relay. 

Perhaps it's not possible on dual r2. 


Regards,

**Reply by barbudor** [reply]
Did you used `switchmode114  1` to enable the option ?

**Reply by dokho33** [reply]
SwitchMode114 not exist, but i've try with setoption114 and don't work

**Reply by barbudor** [reply]
Yes I meant `Setoption114  1`

Did you set it to 1 ?
It works fine here

```
20:54:43.789 CMD: rule1
20:54:43.798 MQT: stat/nodemcu/RULE = {"Rule1":"ON","Once":"OFF","StopOnError":"OFF","Length":41,"Free":470,"Rules":"on switch1#state=3 do power1 toggle endon"}
20:54:48.964 CMD: setoption114 1
20:54:48.971 MQT: stat/nodemcu/SETOPTION = {"SetOption114":"ON"}
20:54:53.512 MQT: stat/nodemcu/SWITCH1 = {"Switch1":{"Action":"TOGGLE"}}
20:55:00.745 RUL: SWITCH1#STATE=3 performs "power1 toggle"
20:55:00.753 MQT: stat/nodemcu/POWER = {"POWER":"ON"}
20:55:00.757 MQT: stat/nodemcu/POWER = ON (retained)
20:55:00.764 MQT: stat/nodemcu/SWITCH1 = {"Switch1":{"Action":"HOLD"}}
```

**Reply by dokho33** [reply]
Yes. I set it to 1 but relay is still toggled by switch.

Works fine for you on a sonoff dual r2 ?

**Reply by barbudor** [reply]
Not on a R2 but that should be independant
Can you share the result of `template` and `gpio` ?

**Reply by dokho33** [reply]
```
[22:11:02.320 CMD: Groupe 0, Index 1, Commande "TEMPLATE", Donnée ""
22:11:02.327 MQT: stat/tasmota_98C5B6/RESULT = {"NAME":"Generic","GPIO":[32,3072,0,3104,0,0,33,0,224,320,0,0,0,0],"FLAG":0,"BASE":43}
```
```
22:13:13.872 CMD: Groupe 0, Index 1, Commande "GPIO", Donnée ""
22:13:13.878 MQT: stat/tasmota_98C5B6/RESULT = {"GPIO0":{"32":"Bouton1"},"GPIO1":{"0":"Aucun"},"GPIO3":{"0":"Aucun"},"GPIO9":{"33":"Bouton2"}}
```

**Reply by barbudor** [reply]
There is no switches in your configuration but buttons. If you want to decouple Buttons from relay then the command is not `Setoption114` but `Setoption73`
switches and buttons are different and behavior are different

Beside you are talking about a Donoff Dual T2 but your template is for Sonoff POW R2

So please double check which device you are talking about and if you want to use switches or buttons

**Reply by dokho33** [reply]
Hum ... i'm dumb ... works fine with setoption73. Thank you.

You say my template is for POW R2 but in "module configuration" menu i've set "Sonoff Dual R2 (39)" ...

**Reply by barbudor** [reply]
The template you given is probably not activated then

Bonne continuation

**Reply by dokho33** [reply]
Thank you :)

PS: are you French?
